clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2882,12 +2883,13 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     // (of either the lambda or the function object).
     CXXRecordDecl *KernelObj =
         GetSYCLKernelObjectType(KernelCallerFunc)->getAsCXXRecordDecl();
-    CXXMethodDecl *WGLambdaFn = nullptr;
-    if (KernelObj->isLambda())
-      WGLambdaFn = KernelObj->getLambdaCallOperator();
-    else
-      WGLambdaFn = getOperatorParens(KernelObj);
-    assert(WGLambdaFn && "non callable object is passed as kernel obj");
+
+    KernelCallOperatorVisitor KernelCallOperator(KernelCallerFunc, KernelObj);
+    CXXMethodDecl *WGCallOperator = nullptr;
+
+    WGCallOperator = CallOperator;
```

Comment:
I suggest removing these unnecessary lines and use CallOperator instead of WGCallOperator later in the code.
```suggestion
```